core/domain/classifier_services.py

```diff
@@ -127,6 +127,61 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def handle_jobs_creation(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration domain object.
+        state_names: list(str). List of state names.
+    """
+    job_dicts_list = []
+    exp_id = exploration.id
+    exp_version = exploration.version
+    states = exploration.states
+    for state_name in state_names:
+        state = states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        job_dicts_list.append({
+            'algorithm_id': algorithm_id,
+            'interaction_id': interaction_id,
+            'exp_id': exp_id,
+            'exp_version': exp_version,
+            'state_name': state_name,
+            'training_data': training_data,
+            'status': feconf.TRAINING_JOB_STATUS_NEW
+        })
+    create_multi_classifier_training_jobs(job_dicts_list)
+
+
+def handle_mappings_creation(exploration, state_names, change_list):
+    """Creates new TrainingJobExplorationMappingModel instances for all the
+    state names passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration domain object.
+        state_names: list(str). List of state names.
+        change_list: list(dict). A list of changes introduced in this commit.
+    """
+    exp_id = exploration.id
+    exp_version = exploration.version
+    for state_name in state_names:
+        old_state_name = state_name
+        classifier_training_job = get_job(exp_id, exp_version-1, old_state_name)
```

Comment:
set `exp_version - 1` to old_exp_version somewhere and use it.

Also add a little comment explaining that we are mapping old state's model to a new state.